hardware_interface/include/joint_limits/joint_limits_rosparam.hpp

```diff
@@ -14,17 +14,60 @@
 
 /// \author Adolfo Rodriguez Tsouroukdissian
 
-#ifndef JOINT_LIMITS_INTERFACE__JOINT_LIMITS_ROSPARAM_HPP_
-#define JOINT_LIMITS_INTERFACE__JOINT_LIMITS_ROSPARAM_HPP_
-
-#include <joint_limits_interface/joint_limits.hpp>
-
-#include <rclcpp/rclcpp.hpp>
+#ifndef JOINT_LIMITS__JOINT_LIMITS_ROSPARAM_HPP_
+#define JOINT_LIMITS__JOINT_LIMITS_ROSPARAM_HPP_
 
+#include <limits>
 #include <string>
 
-namespace joint_limits_interface
+#include "joint_limits/joint_limits.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace joint_limits
+{
+inline bool declare_parameters(const std::string & joint_name, const rclcpp::Node::SharedPtr & node)
 {
+  const std::string param_base_name = "joint_limits." + joint_name;
+  try
+  {
+    node->declare_parameter<bool>(param_base_name + ".has_position_limits", false);
```

Comment:
I think it might be good to check whether the parameter is declared or not, before declaring it. I've encountered errors after using automatically_declare_parameters_from_overrides(true) for the node I've used to read the joint limits